packages/client/tests/functional/concurrent-transaction-unique/tests.ts

```diff
@@ -0,0 +1,45 @@
+// @ts-ignore
+import { PrismaClient } from '@prisma/client'
+import crypto from 'crypto'
+
+import testMatrix from './_matrix'
+
+declare let prisma: PrismaClient
+
+const data = [
+  { name: 'name', id: crypto.randomBytes(12).toString('hex') },
+  { name: 'name', id: crypto.randomBytes(12).toString('hex') },
+  { name: 'name', id: crypto.randomBytes(12).toString('hex') },
+]
+
+/**
+ * Reproduction for issue #9678
+ */
+testMatrix.setupTestSuite(({ provider }) => {
+  testIf(provider !== 'sqlite')('serialized deleteMany/createMany', async () => {
+    const fn = async () => {
+      await prisma.$transaction([
+        prisma.resource.deleteMany({ where: { name: 'name' } }),
+        prisma.resource.createMany({ data }),
+      ])
+    }
+
+    for (let i = 0; i < 50; i++) {
+      await fn()
+    }
```

Comment:
This is meant to test that one can run this a lot of times serially without it failing?

Reply:
Yes, the user said it also fails that way, which was suprising but I tested it anyways.

Reply:
https://github.com/prisma/prisma/pull/15162/files#r980403641